handsontable/src/plugins/undoRedo/undoRedo.js

```diff
@@ -457,6 +457,37 @@ UndoRedo.ChangeAction.prototype.undo = function(instance, undoneCallback) {
     }
   }
 
+  const selectedLast = instance.getSelectedLast();
+
+  if (selectedLast !== undefined) {
+    const [changedRow, changedColumn] = data[0];
+    const [selectedRow, selectedColumn] = selectedLast;
+    const firstFullyVisibleRow = instance.view.getFirstFullyVisibleRow();
+    const firstFullyVisibleColumn = instance.view.getFirstFullyVisibleColumn();
+    const isInVerticalViewPort = changedRow >= firstFullyVisibleRow;
+    const isInHorizontalViewPort = changedColumn >= firstFullyVisibleColumn;
+    const isInViewport = isInVerticalViewPort && isInHorizontalViewPort;
+    const isChangedSelection = selectedRow !== changedRow || selectedColumn !== changedColumn;
+
+    // Performing scroll only when selection has been changed right after editing a cell.
+    if (isInViewport === false && isChangedSelection === true) {
```

Comment:
I'm not sure about the `isChangedSelection` condition. 
It _is_ in line [with this comment](https://github.com/handsontable/dev-handsontable/issues/115#issuecomment-1833436440), but does not meet the requirements of the original post of https://github.com/handsontable/dev-handsontable/issues/115 (it does not require selecting a different cell before hitting `cmd+z`)

Is this what we want to do with it? 
cc: @evanSe

Reply:
@jansiegel the current implementation in line with the comment is correct, @wszymanski we want a code sandbox for @krzysztofspilka review before this is merged to develop.

Reply:
> @jansiegel the current implementation in line with the comment is correct, @wszymanski we want a code sandbox for @krzysztofspilka review before this is merged to develop.

Here you go: https://jsfiddle.net/916z0tvc

Reply:
@wszymanski we are good to merge